### ISSUE TITLE
UnicodeEncodeError: 'ascii' codec can't encode character u'\xa0' in position 480: ordinal not in range(128)

### ISSUE THREAD

**Issue by issmirnov** [chrisallenlane/cheat#372]
Piping cheat output for certain commands fails.

```
# cheat find | grep "xargs"`
Alias tip: c find | grep "xargs"
Traceback (most recent call last):
  File "/usr/local/bin/cheat", line 65, in <module>
    print(colorize(sheet.read(options['<cheatsheet>'])))
UnicodeEncodeError: 'ascii' codec can't encode character u'\xa0' in position 480: ordinal not in range(128)
```

Full paste of `find` cheat.

```
# To find files by case-insensitive extension (ex: .jpg, .JPG, .jpG):
find . -iname "*.jpg"

# To find directories:
find . -type d

# To find files:
find . -type f

# To find files by octal permission:
find . -type f -perm 777

# To find files with setuid bit set:
find . -xdev \( -perm -4000 \) -type f -print0 | xargs -0 ls -l

# To find files with extension '.txt' and remove them:
find ./path/ -name '*.txt' -exec rm '{}' \;

# To find files with extension '.txt' and look for a string into them:
find ./path/ -name '*.txt' | xargs grep 'string'

# To find files with size bigger than 5 Mebibyte and sort them by size:
find . -size +5M -type f -print0 | xargs -0 ls -Ssh | sort -z

# To find files bigger than 2 Megabyte and list them:
find . -type f -size +200000000c -exec ls -lh {} \; | awk '{ print $9 ": " $5 }'

# To find files modified more than 7 days ago and list file information
find . -type f -mtime +7d -ls

# To find symlinks owned by a user and list file information
find . -type l --user=username -ls

# To search for and delete empty directories
find . -type d -empty -exec rmdir {} \;

# To search for directories named build at a max depth of 2 directories
find . -maxdepth 2 -name build -type d

# To search all files who are not in .git directory
find . ! -iwholename '*.git*' -type f

# To find all files that have the same node (hard link) as MY_FILE_HERE
find . -type f -samefile MY_FILE_HERE 2>/dev/null

# To find all files in the current directory and modify their permissions
find . -type f -exec chmod 644 {} \;

```

Other commands work fine, ie `cheat nmap | grep a` doesn't crash.

**Comment by issmirnov** [comment]
Looks like adding `string.replace(u'\xa0', u' ')` might solve this.

**Comment by yumitsu** [comment]
I fixed this by modifying `bin/cheat` at line 65:
```python
print(colorize(sheet.read(options['<cheatsheet>'])).encode('utf-8'))
```

**Comment by Sicabol** [comment]
Same issue, without being able to change the file as it is in a snap.

**Comment by bernermic** [comment]
> without being able to change the file as it is i

I created the snap version and I fixed this in the snap via patching the encoding into the snap. Please have a look if it is working with the latest version (`snap refresh cheat`).

**Comment by chrisallenlane** [comment]
This should be resolved as of changes that were merged in today with #382. These changes should be deployed in the upcoming days with a new release. With that said, I'll close this as resolved.

Thanks for your help.
